kernel/Base/Element.h

```diff
@@ -74,7 +76,9 @@ class PhysicalElement;
 
 // class is final as a reminder that the virtual default destructor should be
 // added once something inherits from this class
-class Element final : public Geometry::ElementGeometry, public ElementData {
+class Element final : public Geometry::ElementGeometry,
```

Comment:
So I am not a fan of multiple inheritance the new baseclass you added is admittedly the best candidate of the 3.

Reply:
The hierarchy here is kinda strange. Both ElementGeometry is just the geometric information about the element, and from that point of view should either be a field or just inlined. ElementData is similar, except that it has quite a bit of data that should, in my view, not be in Element at all.

The new MeshEntity would indeed be better as base class.